packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Models/Types/JsonMrwSerializationTypeProvider.cs

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Linq;
+using System.Text.Json;
+using Microsoft.Generator.CSharp.ClientModel.Expressions;
+using Microsoft.Generator.CSharp.Expressions;
+
+namespace Microsoft.Generator.CSharp.ClientModel
+{
+    /// <summary>
+    /// This class provides the set of serialization models, methods, and interfaces for a given model.
+    /// </summary>
+    internal sealed class JsonMrwSerializationTypeProvider : TypeProvider
+    {
+        private readonly Parameter SerializationOptionsParameter =
+            new("options", null, typeof(ModelReaderWriterOptions), null, ValidationType.None, null);
+        private readonly CSharpType _iJsonModelTInterface;
+        private readonly CSharpType? _iJsonModelObjectInterface;
+        private readonly CSharpType _iPersistableModelTInterface;
+        private readonly CSharpType? _iPersistableModelObjectInterface;
+
+        public JsonMrwSerializationTypeProvider(ModelTypeProvider model) : base(null)
+        {
+            Model = model;
+            Name = model.Name;
+            // Initialize the serialization interfaces
+            _iJsonModelTInterface = new CSharpType(typeof(IJsonModel<>), Model.Type);
+            _iJsonModelObjectInterface = Model.IsStruct ? (CSharpType)typeof(IJsonModel<object>) : null;
+            _iPersistableModelTInterface = new CSharpType(typeof(IPersistableModel<>), Model.Type);
+            _iPersistableModelObjectInterface = Model.IsStruct ? (CSharpType)typeof(IPersistableModel<object>) : null;
+        }
+
+        public override string Name { get; }
+        internal ModelTypeProvider Model { get; }
```

Comment:
it doesn't look like this is used outside of the ctor for this class does it need to be an internal property?

Reply:
I'll make it a field.